caffe2/onnx/onnx_exporter.cc

```diff
@@ -768,6 +772,79 @@ ConvertedResult OnnxExporter::CreateChannelShuffleNodes(
   return result;
 }
 
+ConvertedResult OnnxExporter::CreateReduceMeanNodes(
+    const caffe2::OperatorDef& def,
+    const std::unordered_map<std::string, caffe2::TensorShape>& shapes) {
+    CAFFE_ENFORCE_GE(def.input_size(), 1);
+    CAFFE_ENFORCE_LE(def.input_size(), 2);
+    CAFFE_ENFORCE_EQ(def.input_size(), 1, "Input \"lengths\" is not supported.");
+    CAFFE_ENFORCE_GE(def.output_size(), 1);
+    const auto& x = def.input(0);
+    const auto& y = def.output(0);
+    const auto& dims = shapes.at(x).dims();
+
+    ConvertedResult result;
+    auto& nodes = result.first;
+    auto& const_tensors = result.second;
+    std::unordered_map<std::string, const caffe2::Argument*> args;
+    for (const auto& a : def.arg()) {
+        args.emplace(a.name(), &a);
+    }
+
+    std::vector<int64_t> axes;
+    int64_t keepdims = 1;
+
+    if (def.type() == "ReduceMean") {
+        // axes
+        {
+            const auto it = args.find("axes");
+            if (it == args.end()) {
+                axes.resize(dims.size());
+                std::iota(axes.begin(), axes.end(), 0);
+            } else {
+                axes.assign(it->second->ints().begin(), it->second->ints().end());
+            }
+        }
+
+        // keepdims
+        {
+            const auto it = args.find("keepdims");
+            if (it != args.end()) {
+                keepdims = it->second->i();
+            }
+        }
+    } else {
+        // num_reduce_dim
+        {
```

Comment:
Also, the extra block is useless here. And it will make the coding style so inconsistent...